### PR TITLE
fix: make to_pandas override enable_downsampling when sampling_method is manually set.

### PR DESCRIPTION
**Pull request by Genesis929**
… is manually set.

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes b/310741105

### CODE REVIEW

File: bigframes/core/blocks.py [hunk 1]
```diff
@@ -454,6 +437,29 @@ def _compute_and_count(
...
+                if bigframes.options.sampling.sampling_method is not None
+                else _UNIFORM
+            )
+        else:
+            enable_downsampling = True
+
+        if random_state is None:
+            random_state = bigframes.options.sampling.random_state
+
+        sampling_method = sampling_method.lower()
```
Comment (jiaxunwu) [inline bigframes/core/blocks.py:456]: Can you put sampling_method logic together?

**Review by jiaxunwu — CHANGES_REQUESTED** [review]

File: bigframes/core/blocks.py [hunk 2]
```diff
@@ -454,6 +437,29 @@ def _compute_and_count(
     ) -> Tuple[pd.DataFrame, int, bigquery.QueryJob]:
         """Run query and download results as a pandas DataFrame. Return the total number of results as well."""
         # TODO(swast): Allow for dry run and timeout.
+        enable_downsampling = bigframes.options.sampling.enable_downsampling
```
[the remark this answers is not in the document]
Comment (Genesis929) [reply bigframes/core/blocks.py]: Done.

File: bigframes/core/blocks.py [hunk 3]
```diff
@@ -454,6 +437,38 @@ def _compute_and_count(
     ) -> Tuple[pd.DataFrame, int, bigquery.QueryJob]:
         """Run query and download results as a pandas DataFrame. Return the total number of results as well."""
         # TODO(swast): Allow for dry run and timeout.
+        enable_downsampling = (
+            True
+            if sampling_method is not None
+            else bigframes.options.sampling.enable_downsampling
+        )
+
+        max_download_size = (
```
Comment (GarrettWu) [inline bigframes/core/blocks.py:446]: Nit: less code:

max_download_size = max_download_size or bigframes.options.sampling.max_download_size

Comment (Genesis929) [reply bigframes/core/blocks.py:446]: Done

**Review by GarrettWu — APPROVED** [review]

**Review by jiaxunwu — APPROVED** [review]

**Pull request merged**